Volley-Inc/eslint-config-volley
# Feature/linting unification

Working on making this config work with the [new flat ESLint format](https://eslint.org/blog/2022/08/new-config-system-part-2/) - which means we don't have to install all the plugins in all of our projects! 

Steps to test: 
- In a target gev plugin, run `npm run lint > lint_orig.txt`
- `npm pack` / link this project to that gev plugin 
- Copy `templates/eslint.config.js` into the plugin
- In the plugin, remove all eslint dev dependencies besides eslint itself (`^8.27` seems to be good)
- Run (or replace npm run lint with) `ESLINT_USE_FLAT_CONFIG=true npx eslint src/ > lint_flat.txt` 
- Compare `lint_flat.txt` to `lint_orig.txt` 


 Right now this branch has a couple shortcomings: 
 - Some linting rules aren't being included. We may have to pick and choose from `eslint-config-airbnb` manually? maybe see what`eslint-auto-config` is pulling in and add those. 
 - Tests (`*.spec.ts`)are not being included. I can't figure out where to add the file wildcards to add it back in.
